src/containers/ListingPage/ListingPage.jsx

```diff
@@ -18,23 +18,48 @@ import ViewBar from './components/ViewBar';
 
 class ListingPage extends Component {
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      viewType: 'grid',
+    };
+    this.onViewChangeToList = this.onViewChangeToList.bind(this);
+    this.onViewChangeToGrid = this.onViewChangeToGrid.bind(this);
+    this.viewType = this.viewType.bind(this);
+  }
+
   componentWillMount() {
-    const { fetchListing, params: { listingId } } = this.props;
-    fetchListing(listingId);
+    const { fetchListingByFilter, params: { listingId } } = this.props;
+    fetchListingByFilter(listingId);
+  }
+
+  onViewChangeToList() {
```

Comment:
Vet ikke om det er en stor ting, men jeg er mer fan av å sende inn type og ha en like metode for å endre samme state, feks:
```javascript
onViewTypeChange(type) {
  this.setState({ viewType: type });
}
```

Da slipper jeg å forholde meg til flere metoder som egentlig er like.